torch/csrc/jit/codegen/cuda/predicate_compute.cpp

```diff
@@ -131,20 +154,6 @@ kir::Bool* PredicateCompute::getInlinePredicate(
   auto root_indices = pred_inds.first;
   bool use_maybe_rfactor = pred_inds.second;
 
-  if (out_tv->getMemoryType() == MemoryType::Local && out_tv->hasReduction() &&
```

Comment:
Did this logic move somewhere? This is trying to catch instances where we're initializing reduction buffers in global and shared memory. We don't want to generate predicates in this case.

Reply:
It was dead code - expr is always a TvOp (we had an explicit check early on, how moved inside firstTvOutput)

And if expr is a tv op, the removed block doesn't do anything, right?

Reply:
This seems strange indeed. I can't remember the reason, it could've been accidental (I was making deeper change in this file). 

Added back, thanks.